pkg/cloud/cloud.go

```diff
@@ -445,3 +446,71 @@ func isBadRequestUpdateInProgress(err error) bool {
 	var badRequest *types.BadRequest
 	return errors.As(err, &badRequest) && strings.Contains(err.Error(), "Unable to perform the storage capacity update. There is an update already in progress.")
 }
+
+func (c *cloud) FindFileSystemByVolumeName(ctx context.Context, volumeName string) (*FileSystem, error) {
+	var nextToken *string
+	const maxResults = 100
+
+	klog.V(4).InfoS("Searching for existing filesystem", "volumeName", volumeName)
+
+	for {
+		input := &fsx.DescribeFileSystemsInput{
+			MaxResults: aws.Int32(maxResults),
+			NextToken:  nextToken,
+		}
+
+		output, err := c.fsx.DescribeFileSystems(ctx, input)
+		if err != nil {
+			return nil, fmt.Errorf("failed to describe filesystems: %v", err)
+		}
+
+		klog.V(5).InfoS("Checking batch of filesystems", "count", len(output.FileSystems))
+
+		// Search current batch
+		for _, fs := range output.FileSystems {
+			// Skip if filesystem is being deleted
```

Comment:
nit: Let's modify this so that if the lifecycle is not AVAILABLE or CREATING, continue